test/system/src/test/java/io/pravega/test/system/SegmentReaderAPITest.java

```diff
@@ -0,0 +1,517 @@
+        log.info("Next stream cut2 {}", nextStreamCut2);
+
+        assertTrue(nextStreamCut2 != null);
+        assertTrue(nextStreamCut2.asImpl().getPositions().size() == 2);
+        assertTrue(nextStreamCut2.asImpl().getPositions().containsKey(segment1) &&
+                nextStreamCut2.asImpl().getPositions().containsKey(segment2));
+        //Scaling up end
+
+        //Scaling down start
+        ArrayList<Long> toSeal = new ArrayList<>();
+        toSeal.add(computeSegmentId(1, 1));
+        toSeal.add(computeSegmentId(2, 1));
+
+        Map<Double, Double> keyRanges1 = new HashMap<>();
+        keyRanges1.put(0.0, 1.0);
+
+        Boolean status1 = controller.scaleStream(stream,
+                Collections.unmodifiableList(toSeal),
+                keyRanges1,
+                executor).getFuture().get();
+        assertTrue(status1);
+
+        writeEvents(5, writer);
+
+        Segment segment3 = Segment.fromScopedName(streamScope + "/" + streamName + "/3.#epoch.2");
+        log.info("segment3 name :{}", segment3.getScopedName());
+
+        ImmutableMap<Segment, Long> startStreamCut1 = ImmutableMap.of(segment3, 0L);
+        Map<Stream, StreamCut> startSC1 = ImmutableMap.of(Stream.of(streamScope, streamName),
+                new StreamCutImpl(Stream.of(streamScope, streamName), startStreamCut1));
+        log.info("startSC1 :: {} ", startSC1);
+
+        reader0.close();
+        ReaderGroupConfig readerGroupConfig2 = ReaderGroupConfig.builder()
+                .disableAutomaticCheckpoints()
+                .startingStreamCuts(startSC1)
+                .build();
+        readerGroup.resetReaderGroup(readerGroupConfig2);
+
+        //After scale down reading the event
+        reader0 = clientFactory.createReader(readerName,
+                readerGroupName,
+                new UTF8StringSerializer(),
+                ReaderConfig.builder().build());
+        assertEquals(readEvent(reader0), 5);
+        reader0.close();
+
+        ArrayList<SegmentRange> rangeList3 = Lists.newArrayList(batchClient.getSegments(stream, StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+        Map<Segment, Long> map = rangeList3.stream().collect(
+                Collectors.toMap(SegmentRange::getSegment, SegmentRange::getEndOffset));
+
+        // Getting streamcut at the tail of segment 1 and segment 2
+        StreamCut streamCut = new StreamCutImpl(stream,
+                ImmutableMap.of(segment1, map.get(segment1), segment2, map.get(segment2)));
+
+        StreamCut nextStreamCut3 = batchClient.getNextStreamCut(streamCut, 180L);
+        log.info("Next stream cut3 {}", nextStreamCut3);
+
+        assertTrue(nextStreamCut3 != null);
+        assertTrue(nextStreamCut3.asImpl().getPositions().size() == 1);
+        assertTrue(150L <= nextStreamCut3.asImpl().getPositions().get(segment3).longValue());
+    }
+
+    private URI fetchControllerURI() {
```

Comment:
This creates a new controller service and returns its uri. It is being used as a getter, and is invoked 3 times in the test.